tsl/src/reorder.c

```diff
@@ -148,12 +149,38 @@ tsl_move_chunk(PG_FUNCTION_ARGS)
 				 errmsg("valid chunk, destination_tablespace, and index_destination_tablespaces "
 						"are required")));
 
+	chunk = ts_chunk_get_by_relid(chunk_id, false);
+
+	if (NULL == chunk)
+		ereport(ERROR,
+				(errcode(ERRCODE_INVALID_PARAMETER_VALUE),
+				 errmsg("\"%s\" is not a chunk", get_rel_name(chunk_id))));
+
+	if (ts_chunk_contains_compressed_data(chunk))
+		ereport(ERROR,
+				(errcode(ERRCODE_FEATURE_NOT_SUPPORTED),
+				 errmsg("\"%s\" is a compressed chunk", get_rel_name(chunk_id)),
+				 errhint("Please use the corresponding chunk on the uncompressed hypertable "
+						 "instead.")));
```

Comment:
This can be a potentially confusing error. Since the user can call `compress_chunk('A')`, I guess the user considers `A` to be compressed. But this error refers to the internal chunk (let's call it `B`) associated with `A` as compressed. It is also a bit odd to refer to the main hypertable as being "uncompressed". I don't think that's how the user thinks of the table given that this is the table they enabled compression on.

I think we need to be clear in the error that the operation was executed on an internal chunk and refer to `A`. For instance:

```suggestion
		ereport(ERROR,
				(errcode(ERRCODE_FEATURE_NOT_SUPPORTED),
				 errmsg("cannot directly move internal compression data"),
				 errdetail("Chunk \"%s\" contains compressed data for chunk \"%s\" and cannot be moved directly.", get_rel_name(chunk_B), get_rel_name(chunk_A)),
				 errhint("Moving chunk \"%s\" will also move the compressed data.",
				 get_rel_name(chunk_A))));
```